scripts/yggdrasil-iptunnel/yggdrasil-setup

```diff
@@ -4,36 +4,137 @@
+if [[ -z "${IPv6SUBNETINT}" ]]; then
+    IPv6SUBNETINT="wlan-ap"
+fi
 
 if [ -e /etc/yggdrasil.iptunnel.server ]; then
-    # Add route for cjdns ygg0 interface
-    sudo route add -net "${SUBNET4}0/24" ygg0 || true
 
+
+    SUBNET4=$(confget -f "/etc/yggdrasil.iptunnel.conf" -s "iptunnel" SUBNET4)
+    if [[ -z "${SUBNET4}" ]]; then
+        SUBNET4="10.10.0.0/24"
+    fi
+    SUBNET6=$(confget -f "/etc/yggdrasil.iptunnel.conf" -s "iptunnel" SUBNET6)
+    if [[ -z "${SUBNET6}" ]]; then
+        SUBNET6="fd00::/64"
+    fi
+    OUTINT="$(confget -f "/etc/yggdrasil.iptunnel.conf" -s "iptunnel" outint)"
+    if [[ -z "${OUTINT}" ]]; then
+        OUTINT="eth0"
+    fi
+
+    while read -r PUBLIC_KEY IP_ADDR IP6_ADDR IP6_SUBNET; do
+        if [[ "${PUBLIC_KEY}" =~ ^[0-z]{64} ]]; then
+            yggdrasilctl addRoute subnet="${IP_ADDR}/32" box_pub_key="${PUBLIC_KEY}"
+            if [[ ! -z "${IP6_ADDR}" ]]; then
+                yggdrasilctl addRoute subnet="${IP6_ADDR}/128" box_pub_key="${PUBLIC_KEY}"
+                if [[ ! -z "${IP6_SUBNET}" ]]; then
+                    yggdrasilctl addRoute subnet="${IP6_SUBNET}" box_pub_key="${PUBLIC_KEY}"
+                    ip -6 route add "${IP6_SUBNET}" dev "${YGGINT}" || true
+                fi
+                isIPv6=true
+            fi
+        fi
+    done < /etc/yggdrasil.iptunnel.server
+
+    yggdrasilctl addSourceSubnet subnet="0.0.0.0/0"
+    ip route add "${SUBNET4}" dev "${YGGINT}" || true
+
+    if [[ "${isIPv6}" == true ]]; then
+        yggdrasilctl addSourceSubnet subnet="::/0"
+        # Add route for ygg0 interface
+        ip -6 route add "${SUBNET6}" dev "${YGGINT}" || true
+    fi
+
+    # If no NAT (masquarading) being done, start doing it
+    if [ -z "$(iptables -L POSTROUTING -v -n -t nat  | grep MASQUERADE)" ]; then
+        iptables -t nat -A POSTROUTING -o "${OUTINT}" -j MASQUERADE
+    fi
     # If no NAT (masquarading) being done, start doing it
-    if [ -z "$(sudo iptables -L POSTROUTING -v -n -t nat  | grep MASQUERADE)" ]; then
-        sudo iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
+    if [ -z "$(ip6tables -L POSTROUTING -v -n -t nat  | grep MASQUERADE)" ]; then
+        if [[ ! "${IPv6NAT}" == false ]]; then
+            iptables -t nat -A POSTROUTING -o "${OUTINT}" -j MASQUERADE
+        fi
     fi
 # Configure as yggdrasil iptunnel client if client file is present (this is simply a newline-separated list
 # of cjdns public keys in /etc/yggdrasil.iptunnel.client, each key indicating an iptunnel exit server)
 elif [ -e /etc/yggdrasil.iptunnel.client ]; then
     # Add each server to yggdrasil iptunnel connect-to's
-    while read -r PUBLIC_KEY ASSIGNED_IP; do
+    while read -r PUBLIC_KEY IP_ADDR IP6_ADDR IP6_SUBNET; do
         if [[ "${PUBLIC_KEY}" =~ ^[0-z]{64} ]]; then
-            sudo ip addr add dev ygg0 ${ASSIGNED_IP}/32
+            if [[ ! -z "${IP6_ADDR}" ]]; then
+                ip addr add dev "${YGGINT}" "${IP6_ADDR}/128"
```

Comment:
How do you know it's `/128`?

Reply:
/128 is the   /32 of ipv6

single ip address

Reply:
Thanks, I forget it was CIDR.